src/server/services/home.ts

```diff
@@ -99,7 +99,7 @@ async function fetchCommercial(
 export async function fetchMyLocation(
```

Comment:
Wat gebeurt er in deze functie? Ik merk aan de returnType dat er iets mis gaat. Volgens mij kan dit een stuk simpeler maar ben benieuwd of ik iets mis?

```suggestion
const {content: commercialAddresses} = await fetchCommercial(
  requestID,
  authProfileAndToken
);

if (authProfileAndToken.profile.profileType === 'commercial') {
  return apiSuccessResult(commercialAddresses.filter((location) => location !== null))
}

const {content: privateAddresses} = await fetchPrivate(
  requestID,
  authProfileAndToken
);

const locations: BAGData[] = [
  ...(privateAddresses || []),
  ...(commercialAddresses || []),
].filter((location) => location !== null)

if (locations.length === 0) {
  return apiErrorResult('Could not fetch locations.', null);
}

return apiSuccessResult(locations)
```

Reply:
Als je ReturnType kleiner wordt dan wordt ook de logica waarmee de calling functie rekening moet houden simpler/minder.

Reply:
In plaats van unpack or empty array kan je natuurlijk ook nog:
```
const locations = [[privateAddresses], [commercialAddresses]].flat(2).filter(location => location  !== null)
```
Maar ik vind zelf de unpack or empty methode beter te begrijpen